packages/neos-ui-editors/src/Reference/index.js

```diff
@@ -57,6 +61,11 @@ export default class ReferenceEditor extends PureComponent {
             this.setState({isLoading: true});
             this.props.nodeLookupDataLoader.resolveValue(this.getDataLoaderOptions(), this.props.value)
                 .then(options => {
+                    options.map(option => {
+                        option.icon = this.props.nodeTypeRegistry.getNodeType(option.nodeType).ui.icon;
```

Comment:
I'd first get a nodetype and then get it via `$get('ui.icon', nodeType)`. Otherwise you'd need to check if such properties exists.

Reply:
Wait, but how does it acutally work? You do `.map`, but don't assign it to anything.